### PR TITLE
fix(jsonschema): Fixing Get Json Schema error

### PR DESCRIPTION
**Pull request by osalama7**
Fixes JSON namespace creation

note: Json Namespace will be deprecated
- Request to get JSON schema for JSON type namespace is not necessary 
- Return {} for empty format field
- Validated with Integration tests

### CODE REVIEW

File: idm/meta/json_schema/json_schema.go [hunk 1]
```diff
@@ -350,6 +350,9 @@ func GetJsonSchema(label string, format string) ([]byte, error) {
 }
 
 func GetJsonSchemaSample(label string, name string, format string) (*structpb.Struct, error) {
+	if format == "" {
```
Comment (cristianoliveira) [inline idm/meta/json_schema/json_schema.go:353]: This looks like a regression: `Format` seems only relevant for date/time schemas, but this now returns `nil` for all field types when `Format` is omitted. Is that intended?

   For example, `GetJsonSchemaSample("string", "my_namespace", "")` should still return a valid schema.

Could we add a regression test? `withDateTimeSchema()` already defaults empty/unknown format to `date-time`.

 ```go
   Convey("GetJsonSchemaSample should default empty format for date fields", func() {
      schema, err := GetJsonSchemaSample("date", "my_date_namespace", "")

      So(err, ShouldBeNil)
      So(schema, ShouldNotBeNil)

      properties := schema.GetFields()["properties"].GetStructValue().GetFields()
      dateSchema := properties["my_date_namespace"].GetStructValue().GetFields()

      So(dateSchema["format"].GetStringValue(), ShouldEqual, "date-time")
   })
 ```

File: idm/meta/rest/handler.go [hunk 2]
```diff
@@ -289,13 +289,15 @@ func (s *UserMetaHandler) GetFieldSchema(req *restful.Request, rsp *restful.Resp
...
+	fd := req.QueryParameter("FieldType")
+	ns := req.QueryParameter("Namespace")
+	fp := req.QueryParameter("Format")
+
+	schema, err := s.ServiceClient(ctx).GetNamespaceSchema(ctx, &idm.GetNamespaceSchemaRequest{FieldType: fd, Namespace: ns, Format: fp})
 
 	if err != nil {
-		return err
+		log.Logger(ctx).Error("Error getting namespace schema", zap.String("fieldType", fd), zap.String("namespace", ns), zap.Error(err))
+		return errors.WithMessagef(errors.UnmarshalError, "failed to generate schema for namespace %s: %v", ns, err)
```
Comment (cristianoliveira) [inline idm/meta/rest/handler.go:300]: Minor: 
This basically converts every `GetNamespaceSchema` failure into `UnmarshalError`, which may hide the original error/status. I believe the original error type should be preserved

File: idm/meta/json_schema/json_schema.go (hunk 1 above)
Comment (osalama7) [reply idm/meta/json_schema/json_schema.go:353]: I validated the API responses for all combinations with JSON schema [snapshots](https://github.com/pydio/cells-integration-tests/tree/main/tests/integration/metadata/__snapshots__)

```
go run main.go test ./tests/integration/metadata/... --tags=storage --verbose
=== DEBUG GetSuiteConfig
=== INFO  no config ID found, spinning a new test config
=== DEBUG StartStorage
=== INFO  Opening BoltTDB from /Users/osama.salama/Documents/cells-integration-tests/exec/livid/test-suite-index.db and exposing on port 10222
=== DEBUG NewSuiteRun
=== WARN  no bucket named test-objects can be found in local bolt db
=== DEBUG Runconfig lividdream created
# PYDIO CELLS INTEGRATION TESTS

### Launching "test" sub-command:
      Target server: https://localhost:8080
      ConfigID: livid
      RunID: lividdream
      Version: v4.9.95-dev


### SETUP PHASE
        export CELLS_IT_RUN_ID=lividdream; export CELLS_IT_WORKING_DIR=/Users/osama.salama/Documents/cells-integration-tests/exec;
        /usr/local/go/bin/go test ./environment/setup_test.go -v -parallel 2 -short -tags storage -timeout 45m
=== RUN   TestSetupEnvironment
=== PAUSE TestSetupEnvironment
=== CONT  TestSetupEnvironment
=== RUN   TestSetupEnvironment/TestGlobalSetup
=== PAUSE TestSetupEnvironment/TestGlobalSetup
=== CONT  TestSetupEnvironment/TestGlobalSetup
... Configuring pydiods1
       pydiods1 : OK
--- PASS: TestSetupEnvironment (0.00s)
    --- PASS: TestSetupEnvironment/TestGlobalSetup (0.72s)
PASS
ok      command-line-arguments  0.998s

### REAL TESTS
        export CELLS_IT_RUN_ID=lividdream; export CELLS_IT_WORKING_DIR=/Users/osama.salama/Documents/cells-integration-tests/exec;
        /usr/local/go/bin/go test ./tests/integration/metadata/... -v -parallel 2 -short -tags storage -timeout 45m
=== RUN   TestNodeMetadata
=== PAUSE TestNodeMetadata
=== RUN   TestMetadata
=== PAUSE TestMetadata
=== RUN   TestUserMeta
=== PAUSE TestUserMeta
=== CONT  TestNodeMetadata
=== CONT  TestUserMeta
=== RUN   TestNodeMetadata/SkipTestMetaEntityUnlinkOnClear
=== RUN   TestUserMeta/SkipTestUserMeta128KJsonValue
=== NAME  TestNodeMetadata/SkipTestMetaEntityUnlinkOnClear
    test_case.go:55: 
        
=== NAME  TestUserMeta/SkipTestUserMeta128KJsonValue
    test_case.go:55: 
        
=== RUN   TestNodeMetadata/TestAppendMetaToNode
=== RUN   TestUserMeta/SkipTestUserMetaNameSpace
=== PAUSE TestNodeMetadata/TestAppendMetaToNode
=== RUN   TestNodeMetadata/TestTagCloudEntityLinkingMultipleFiles
=== NAME  TestUserMeta/SkipTestUserMetaNameSpace
    test_case.go:55: 
        
=== RUN   TestUserMeta/SkipTestUserMetaTag
=== PAUSE TestNodeMetadata/TestTagCloudEntityLinkingMultipleFiles
=== NAME  TestUserMeta/SkipTestUserMetaTag
    test_case.go:55: 
        
=== RUN   TestUserMeta/SkipTestUserMetaTagOnNode
=== RUN   TestNodeMetadata/TestTagCloudWithEntityItems
=== NAME  TestUserMeta/SkipTestUserMetaTagOnNode
    test_case.go:55: 
        
=== PAUSE TestNodeMetadata/TestTagCloudWithEntityItems
--- PASS: TestUserMeta (0.00s)
    --- SKIP: TestUserMeta/SkipTestUserMeta128KJsonValue (0.00s)
    --- SKIP: TestUserMeta/SkipTestUserMetaNameSpace (0.00s)
    --- SKIP: TestUserMeta/SkipTestUserMetaTag (0.00s)
    --- SKIP: TestUserMeta/SkipTestUserMetaTagOnNode (0.00s)
=== CONT  TestNodeMetadata/TestAppendMetaToNode
=== CONT  TestMetadata
=== RUN   TestMetadata/TestAdminOnlyNamespace
=== PAUSE TestMetadata/TestAdminOnlyNamespace
=== RUN   TestMetadata/TestLegacyDefinitionsCRUD
=== PAUSE TestMetadata/TestLegacyDefinitionsCRUD
=== RUN   TestMetadata/TestSchemaEndpointCombinations
=== PAUSE TestMetadata/TestSchemaEndpointCombinations
=== RUN   TestMetadata/TestSchemaParam
=== PAUSE TestMetadata/TestSchemaParam
=== RUN   TestMetadata/TestStandardUserCRUD
=== PAUSE TestMetadata/TestStandardUserCRUD
=== CONT  TestNodeMetadata/TestTagCloudWithEntityItems
=== CONT  TestNodeMetadata/TestTagCloudEntityLinkingMultipleFiles
=== CONT  TestMetadata/TestAdminOnlyNamespace
=== CONT  TestMetadata/TestSchemaParam
=== CONT  TestMetadata/TestStandardUserCRUD
=== RUN   TestMetadata/TestStandardUserCRUD/Get_global_namespace_schema_and_assert
=== RUN   TestMetadata/TestStandardUserCRUD/Get_global_namespace_schema_and_assert#01
=== RUN   TestMetadata/TestStandardUserCRUD/Get_global_namespace_schema_and_assert#02
=== RUN   TestMetadata/TestStandardUserCRUD/Get_global_namespace_schema_and_assert#03
=== RUN   TestMetadata/TestStandardUserCRUD/Get_global_namespace_schema_and_assert#04
=== RUN   TestMetadata/TestStandardUserCRUD/Get_global_namespace_schema_and_assert#05
=== RUN   TestMetadata/TestStandardUserCRUD/Get_global_namespace_schema_and_assert#06
=== RUN   TestMetadata/TestStandardUserCRUD/Get_global_namespace_schema_and_assert#07
=== RUN   TestMetadata/TestStandardUserCRUD/Get_global_namespace_schema_and_assert#08
=== RUN   TestMetadata/TestStandardUserCRUD/Get_global_namespace_schema_and_assert#09
=== RUN   TestMetadata/TestStandardUserCRUD/Get_global_namespace_schema_and_assert#10
=== CONT  TestMetadata/TestSchemaEndpointCombinations
=== CONT  TestMetadata/TestLegacyDefinitionsCRUD
--- PASS: TestMetadata (0.00s)
    --- PASS: TestMetadata/TestAdminOnlyNamespace (0.28s)
    --- PASS: TestMetadata/TestSchemaParam (0.08s)
    --- PASS: TestMetadata/TestStandardUserCRUD (1.07s)
        --- PASS: TestMetadata/TestStandardUserCRUD/Get_global_namespace_schema_and_assert (0.01s)
        --- PASS: TestMetadata/TestStandardUserCRUD/Get_global_namespace_schema_and_assert#01 (0.01s)
        --- PASS: TestMetadata/TestStandardUserCRUD/Get_global_namespace_schema_and_assert#02 (0.01s)
        --- PASS: TestMetadata/TestStandardUserCRUD/Get_global_namespace_schema_and_assert#03 (0.00s)
        --- PASS: TestMetadata/TestStandardUserCRUD/Get_global_namespace_schema_and_assert#04 (0.00s)
        --- PASS: TestMetadata/TestStandardUserCRUD/Get_global_namespace_schema_and_assert#05 (0.01s)
        --- PASS: TestMetadata/TestStandardUserCRUD/Get_global_namespace_schema_and_assert#06 (0.01s)
        --- PASS: TestMetadata/TestStandardUserCRUD/Get_global_namespace_schema_and_assert#07 (0.01s)
        --- PASS: TestMetadata/TestStandardUserCRUD/Get_global_namespace_schema_and_assert#08 (0.01s)
        --- PASS: TestMetadata/TestStandardUserCRUD/Get_global_namespace_schema_and_assert#09 (0.01s)
        --- PASS: TestMetadata/TestStandardUserCRUD/Get_global_namespace_schema_and_assert#10 (0.01s)
    --- PASS: TestMetadata/TestSchemaEndpointCombinations (0.19s)
    --- PASS: TestMetadata/TestLegacyDefinitionsCRUD (0.33s)
--- PASS: TestNodeMetadata (0.00s)
    --- SKIP: TestNodeMetadata/SkipTestMetaEntityUnlinkOnClear (0.00s)
    --- PASS: TestNodeMetadata/TestAppendMetaToNode (5.81s)
    --- PASS: TestNodeMetadata/TestTagCloudWithEntityItems (7.79s)
    --- PASS: TestNodeMetadata/TestTagCloudEntityLinkingMultipleFiles (12.84s)
PASS
ok      github.com/pydio/cells-integration-tests/tests/integration/metadata     18.922s

### TEARDOWN PHASE
        export CELLS_IT_RUN_ID=lividdream; export CELLS_IT_WORKING_DIR=/Users/osama.salama/Documents/cells-integration-tests/exec;
        /usr/local/go/bin/go test ./environment/teardown_test.go -v -parallel 2 -short -tags storage -timeout 45m
=== RUN   TestTeardownEnvironment
=== PAUSE TestTeardownEnvironment
=== CONT  TestTeardownEnvironment
=== RUN   TestTeardownEnvironment/TestGlobalTeardown
=== PAUSE TestTeardownEnvironment/TestGlobalTeardown
=== CONT  TestTeardownEnvironment/TestGlobalTeardown
### Cleaning main test groups
       - /testers
    
--- PASS: TestTeardownEnvironment (0.00s)
    --- PASS: TestTeardownEnvironment/TestGlobalTeardown (3.26s)
PASS
ok      command-line-arguments  3.543s
  All run launched, sleeping 3 seconds before closing the server.
  All tests completed, storage closed, and HTTP server stopped.

### Tests against https://localhost:8080 **DONE** in 1m10s.
```

**Review by cristianoliveira — APPROVED** [review]
